### PR TITLE
Fix CPU_COUNT check for FreeBSD 13.1+

### PR DESCRIPTION
**Pull request by Thefrank**
HAVE_GNU_CPU_COUNT has always incorrectly failed for FreeBSD. 

This has not been an issue until FreeBSD 13.1 with its addition of `sched_getaffinity` and `sched_setaffinity`.

### CODE REVIEW

**Comment by Thefrank** [comment]
more info: With the two sched checks passing but CPU_COUNT failing it will try and make its own CPU_COUNT...and fail because FreeBSD already has a macro it uses for that.

File: src/mono/cmake/configure.cmake [hunk 1]
```diff
@@ -155,6 +155,20 @@ check_c_source_compiles(
   "
   HAVE_GNU_STRERROR_R)
 
+if (HOST_FREEBSD)
+check_c_source_compiles(
```
Comment (am11) [inline src/mono/cmake/configure.cmake:159]: nit: we can replace `check_c_source_compiles` with a one liner `check_symbol_exists` to replace both if/else branches (see its usage in this file and other places):

```cmake
check_symbol_exists(CPU_COUNT "sched.h" HAVE_GNU_CPU_COUNT)
```

Comment (Thefrank) [reply src/mono/cmake/configure.cmake]: That is a much better way of doing it

**Comment by Thefrank** [comment]
Switched to `check_symbol_exists` as per @am11 suggestion. 
I left it in the section with the other "GNU" items but for readability it might be better with the other `check_symbol_exists` items

**Review by am11 — APPROVED** [review]
LGTM

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**